### PR TITLE
Set APPEND_FORWARD to forward X-Forwarded-Client-Cert as default

### PR DESCRIPTION
**Pull request by cnvergence**
**Description**
Currently, we are using SANITIZE_SET as default, which does not forward the XFCC header in the requests.

Setting APPEND_FORWARD, we can forward the XFCC header, so when the client connection is mTLS, 
append the client certificate information to the request’s XFCC header and forward it. 

In other cases, behaviour is the same as with SANITIZE_SET.

Changes proposed in this pull request:

- Set APPEND_FORWARD as the new default

See also:
#13492

### CODE REVIEW

**Review by mjakobczyk — DISMISSED** [review]

File: docs/01-overview/main-areas/service-mesh/smsh-02-default-istio-setup-in-kyma.md [hunk 1]
```diff
@@ -32,3 +32,4 @@ These configuration changes are applied to customize Istio for use with Kyma:
 - The `istio-sidecar-injector` Mutating Webhook Configuration is patched to exclude Gardener resources in the `kube-system` Namespace and the timeout is set to 10 seconds.
 - The use of HTTP 1.0 is enabled in the outbound HTTP listeners by `PILOT_HTTP10` flag set in Istiod component environment variables.
 - IstioOperator configuration file is modified. [Change Kyma settings](../../../04-operation-guides/operations/03-change-kyma-config-values.md) to customize the configuration.
+- Forwarding X-Forwarded-Client-Cert header is enabled by `APPEND_FORWARD` setting. This means that when the client connection is mTLS, the client certificate information will be added to the request’s XFCC header. [Istio Gateway settings](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/#configuring-x-forwarded-client-cert-headers)
```
Comment (mmitoraj) [inline docs/01-overview/main-areas/service-mesh/smsh-02-default-istio-setup-in-kyma.md:35]:
```suggestion
- The forwarding of the X-Forwarded-Client-Cert (XFCC) header is enabled using the `APPEND_FORWARD` value. It means that when the client connection is mTLS, the client certificate information is added to the request’s XFCC header. To learn more, read the [official documentation on XFCC headers configuration](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/#configuring-x-forwarded-client-cert-headers).
```

**Review by mmitoraj — COMMENTED** [review]
I added one suggestion :)

**Review by mmitoraj — APPROVED** [review]

**Comment by cnvergence** [comment]
Decided to not change the default value, as the feature request is not valid anymore.

**Pull request closed without merging**